src/gitrepo/manager.go

```diff
@@ -9,18 +9,44 @@ import (
 const RevisionNone = ""
 
 type gitRepoManager struct {
-	worktree *git.Worktree
+	repo *git.Repository
 }
 
 func NewGitRepoManger(repoDir string) gitRepoManager {
 	repo := utils.GetResultOrPanicError(git.PlainOpen(repoDir))
-	worktree := utils.GetResultOrPanicError(repo.Worktree())
-	return gitRepoManager{worktree}
+	return gitRepoManager{repo}
 }
 
 func (m *gitRepoManager) Checkout(refName string) {
-	utils.PanicError(m.worktree.Checkout(&git.CheckoutOptions{
+	worktree := utils.GetResultOrPanicError(m.repo.Worktree())
+	utils.PanicError(worktree.Checkout(&git.CheckoutOptions{
 		Branch: plumbing.ReferenceName("refs/heads/" + refName),
 		Force:  true,
 	}))
 }
+
+func (m *gitRepoManager) ListChangedFiles(fromRevision string, toRevision string) []string {
+	fromHash := utils.GetResultOrPanicError(m.repo.ResolveRevision(plumbing.Revision(fromRevision)))
+	fromCommit := utils.GetResultOrPanicError(m.repo.CommitObject(*fromHash))
+	toHash := utils.GetResultOrPanicError(m.repo.ResolveRevision(plumbing.Revision(toRevision)))
+	toCommit := utils.GetResultOrPanicError(m.repo.CommitObject(*toHash))
+	patch := utils.GetResultOrPanicError(fromCommit.Patch(toCommit))
+	filePathsSet := make(map[string]bool)
+	for _, filePatch := range patch.FilePatches() {
+		fromFile, toFile := filePatch.Files()
+		if fromFile != nil {
+			filePathsSet[fromFile.Path()] = true
+		}
+		filePathsSet[toFile.Path()] = true
```

Comment:
From documentation:
```
	// Files returns the from and to Files, with all the necessary metadata to
	// about them. If the patch creates a new file, "from" will be nil.
	// If the patch deletes a file, "to" will be nil.
```

So, can't toFile also be nil?

Reply:
You're absolutely right. Thanks.